.github/workflows/build_and_tests.yml

```diff
@@ -63,39 +65,77 @@ jobs:
       with:
         fetch-depth: 0
 
-    - name: Get Ragger version
-      id: version
-      run: |
-        echo "##[set-output name=version;]$(cat src/ragger/__init__.py  | grep __version__ | cut -d ' ' -f 3 | sed s/\"//g)";
-        echo "Version registered: ${{ steps.version.outputs.version }}"
-
-    - name: Build Ragger python package
+    - name: Build Ragger Python package
       run: |
         pip install --upgrade pip build twine
         PIP_EXTRA_INDEX_URL=https://test.pypi.org/simple/ python -m build
         python -m twine check dist/*
 
+    - name: Get tag version
+      if: success() && github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags/')
+      id: tag
+      run: |
+        echo "TAG_VERSION=${GITHUB_REF#refs/tags/}" >> "$GITHUB_ENV"
+        echo "Tag version: '${{ env.TAG_VERSION }}'"
+
+    - name: Check version against CHANGELOG
```

Comment:
Maybe we should also check against `__version__` from `src/ragger/__init__.py`?

Reply:
No need if the version is automatically updated from the tag version